python/src/aiconfig/schema.py

```diff
@@ -59,9 +59,23 @@ class PromptMetadata(BaseModel):
     class Config:
         extra = "allow"
 
+class Attachment(BaseModel):
+    '''
+    Attachment used to pass data in PromptInput for non-text inputs (ex: image, audio)
+    '''
+    # The data representing the attachment
+    data: Any
+    # The MIME type of the result. If not specified, the MIME type will be assumed to be plain text.
+    mime_type: Optional[str] = None
+    # Output metadata
+    metadata: Dict[str, Any]
```

Comment:
Something that I'm thinking of when getting GPT-4V working - should attachments have optional names? 

GPT-4V just takes a list of images, but other models could have specific named attachments (the one I'm thinking of is Stable Diffusion with Control net could take one style of control image, a second one, and a prompt - in a model parser might be easier to reference by name rather than by index)

Reply:
Ah, was just thinking something similar when working on the prompt schema. Mainly, parameters right now are really only supporting the prompt.input === "string" scenario, but for attachments we'll need to either support the scenario for input.data === "string" or, better yet, support reference by key so we could potentially also include references to attachments by key as well?

Reply:
I did this for GPT-4V support btw https://github.com/lastmile-ai/lastmile/blob/743d4713bff9663179ec0e2d8508b511fce804d3/src/helpers/client/workbookCellTypes.ts#L45 ; still using an array, but have an optional name so that you can still filter by name (O(n) because array, but at least it's not just indexed).